### PR TITLE
Moving the NodeHostNameIsFQDN amont other nodeset utils

### PR DESCRIPTION
**Pull request by jpodivin**
The file was an aberration from earlier iteration of the project. Getting rid of it and moving the only resident code elsewhere is reasonable.

### CODE REVIEW

**Comment by rabi** [comment]
> The file was an aberration from earlier iteration of the project. Getting rid of it and moving the only resident code elsewhere is reasonable.

No, it was intentional. We should not add all utility functions in *types.go.

**Review by bshephar — COMMENTED** [review]
I'm not sure why or if this is necessary, but it seems Rabi already started that conversation.

I just have a small comment to add in case we do decide to proceed.

File: api/v1beta1/openstackdataplanenodeset_types.go [hunk 1]
```diff
@@ -282,3 +283,13 @@ func (r *OpenStackDataPlaneNodeSetSpec) duplicateNodeCheck(nodeSetList *OpenStac
...
 	return
 }
+
+// NodeHostNameIsFQDN Helper to check if a hostname is fqdn
+func NodeHostNameIsFQDN(hostname string) bool {
+	// Regular expression to match a valid FQDN
+	// This regex assumes that the hostname and domain name segments only contain letters, digits, hyphens, and periods.
+	regex := `^([a-zA-Z0-9-]+\.)*[a-zA-Z0-9-]+\.[a-zA-Z]{2,}$`
+
+	match, _ := regexp.MatchString(regex, hostname)
```
[the remark this answers is not in the document]
Comment (jpodivin) [reply api/v1beta1/openstackdataplanenodeset_types.go:293]: We should. But it wasn't implemented in that way.

**Comment by jpodivin** [comment]
> > The file was an aberration from earlier iteration of the project. Getting rid of it and moving the only resident code elsewhere is reasonable.
> 
> No, it was intentional. We should not add all utility functions in *types.go.

I suppose you know best since you've implemented it. Although I don't understand why we need to have file for one function when other utility functions can live in their respective types files. Nor am I entirely sure why it's in file called `openstackdataplanenodeset.go` when it is used in other places. 

afaik, it was originally in more general location `utils.go`. Why did you move it?

File: api/v1beta1/openstackdataplanenodeset_types.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpodivin) [reply api/v1beta1/openstackdataplanenodeset_types.go:293]: Depends on if we want to move forward with it.

**Comment by rabi** [comment]
> afaik, it was originally in more general location `utils.go`. Why did you move it?

Because it is  now used in the webhook and if you keep it in utils.go it would be create cyclic dependency.

**Comment by jpodivin** [comment]
> > afaik, it was originally in more general location `utils.go`. Why did you move it?
> 
> Because it is now used in the webhook and if you keep it in utils.go it would be create cyclic dependency.

Ok. It wasn't in your PR or commit message, that's why I'm asking. It does make sense to move it then I suppose. Although I'm still not stoked about it having it's own file like that.

**Pull request closed without merging**